### PR TITLE
Python package set: major updates

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Hydra job

https://hydra.nixos.org/jobset/nixpkgs/python-unstable

### CODE REVIEW

**Comment by dotlambda** [comment]
Should I push a libgit2 update to this PR or should we move the pygit2 update to a seperate PR?

**Comment by dotlambda** [comment]
Pytest-flake8 is broken due to pytest bump. See https://github.com/tholo/pytest-flake8/issues/45.

**Comment by dotlambda** [comment]
@FRidh Did you intentionally update python packages that are not located in `pkgs/development/python-modules` such as fava or home-assistant-frontend?

**Comment by FRidh** [comment]
> Should I push a libgit2 update to this PR or should we move the pygit2 update to a seperate PR?

We could, or keep it for the next batch. I think it would be nice to stabilize the listed changes, but feel free to add anyway.

> Pytest-flake8 is broken due to pytest bump. See tholo/pytest-flake8#45.

That package always breaks on every update.

> Did you intentionally update python packages that are not located in pkgs/development/python-modules such as fava or home-assistant-frontend?

I did. We should also keep those updated, though maybe some should be excluded. Typically I go over the list of commits and remove commits I don't actually want. I suppose the home-assistant ones should be excluded?

**Comment by dotlambda** [comment]
> I think it would be nice to stabilize the listed changes

I don't get what you mean.
The updated `pygit2` version requires an updated `libgit2`. Therefore I think the easiest thing to do is putting the `pygit2` update into a seperate PR.

> I suppose the home-assistant ones should be excluded?

Not necessarily. I usually update the frontend when I updae Home Assistant, but it doesn't hurt if it is updated seperately.

File: pkgs/development/python-modules/ftfy/default.nix [hunk 1]
```diff
@@ -11,7 +11,7 @@
 buildPythonPackage rec {
   pname = "ftfy";
 
-  version = "4.4.3";
+  version = "5.3.0";
   # ftfy v5 only supports python3. Since at the moment the only
   # packages that use ftfy are spacy and textacy which both support
   # python 2 and 3, they have pinned ftfy to the v4 branch.
```
Comment (dotlambda) [inline pkgs/development/python-modules/ftfy/default.nix]: @FRidh What are your thoughts about this?

Comment (FRidh) [reply pkgs/development/python-modules/ftfy/default.nix]: Ah, right. This commit needs to be removed. I've accidentally merged this one before.

Comment (FRidh) [reply pkgs/development/python-modules/ftfy/default.nix]: Done. I've force-pushed to the branch.

**Comment by dotlambda** [comment]
Libgit2-glib has not released version 0.27 yet and doesn't compile with lingit 0.27 (cf. https://bugzilla.gnome.org/show_bug.cgi?id=794890).
@FRidh Will you remove the pygit2 bump or should I do that?

**Comment by dotlambda** [comment]
Major packages that are still broken include
- pytest-flake8 (https://github.com/tholo/pytest-flake8/issues/45)
- pytest-sugar (https://github.com/Frozenball/pytest-sugar/issues/134)
- pandas
- (not really major, but I'd personally prefer that not to be broken) vobject (broken due to dateutil bump, see https://github.com/eventable/vobject/pull/111) The applications using it (radicale, papis, khard, ...) could pin the dateutil version as a workaround.

**Comment by dotlambda** [comment]
@FRidh Do you think it's better to make 3.5 the default global protobuf version or is it okay to make it the default for `pythonPackages` only.

**Comment by dotlambda** [comment]
As mentioned in https://github.com/NixOS/nixpkgs/pull/38483/commits/d64839183c2a280d348598eb01960e697b080918, cairosvg should not be updated.

**Comment by FRidh** [comment]
Oops.

I don't really know what to do with protobuf. We can *try* to upgrade the default Python version, but let's see if we can do that outside of this PR.

**Comment by dotlambda** [comment]
I already updated it in https://github.com/NixOS/nixpkgs/pull/38483/commits/4d34eb71b5163c405c4fc5965b86265dff059f97 (included in this PR). It didn't introduce any build failure.

**Comment by FRidh** [comment]
Pandas compatibility with dateutils 2.7 seems still quite far away. I don't think we should break pandas so let's revert the dateutils change.

**Comment by purefn** [comment]
The update from aniso8601 2.0.1 to 3.0.0 broke the tests for flask-restful. Unfortunately there isn't a newer release of flask-restful that is compatible, but there is a [change on the master branch](https://github.com/flask-restful/flask-restful/commit/54979f0a49b2217babc53c5b65b5df10b6de8e05) which fixes the tests. Perhaps we should disable the tests for flask-restful for now?

**Pull request merged**